bench/workgen/workgen.cpp

```diff
@@ -397,57 +461,89 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
-            {
-                const std::lock_guard<std::shared_mutex> lock(*icontext->_dyn_mutex);
-
-                // Add the table into the list of dynamic set.
-                tint_t tint = icontext->_dyn_tint_last;
-                icontext->_dyn_tint[uri] = tint;
-                icontext->_dyn_table_names[tint] = uri;
-                icontext->_dyn_table_runtime[tint] = TableRuntime();
-                ++icontext->_dyn_tint_last;
-                VERBOSE(*_workload, "Created table and added to the dynamic set: " << uri);
+            // Create the table. Simply continue on failure.
+            if (create_table(session, config, uri, mirror_uri, true) == 0) {
+                VERBOSE(*_workload, "Created base table '" << uri << "'");
+                if (_workload->options.mirror_tables) {
+                    // Create the mirror. Retry on failure and throw an exception after
+                    // making too many retry attempts.
+                    config = base_config + uri + "," + BASE_TABLE_APP_METADATA + "false\"";
+                    do
+                        status = create_table(session, config, mirror_uri, uri, false);
+                    while (status == EBUSY && ++retries < TABLE_MAX_RETRIES);
+                    if (status != 0)
+                        THROW_ERRNO(
+                          status, "Failed to create mirror table '" << mirror_uri << "'.");
+                    VERBOSE(*_workload, "Created mirror table '" << mirror_uri << "'");
+                }
+                ++creates;
             }
-            ++creates;
         }
-
         sleep(_workload->options.create_interval);
     }
 
     return 0;
 }
 
+// Randomly select a dynamic table to drop. The selected table is flagged for deletion and
+// added to a pending delete list. If the table has a mirror, the mirror is also flagged for
+// deletion and added to the pending delete list. The function returns an EEXIST error status
+// if the selected table is already scheduled for deletion.
+int
+WorkloadRunner::select_table_for_drop(std::vector<std::string> &pending_delete)
+{
+    // Select a random table to drop. Skip tables already flagged for deletion.
+    ContextInternal *icontext = _workload->_context->_internal;
+    auto table_itr = icontext->_dyn_tint.begin();
+    std::advance(table_itr, workgen_random(_rand_state) % icontext->_dyn_tint.size());
+    if (icontext->_dyn_table_runtime[table_itr->second]._pending_delete) {
+        return EEXIST;
+    }
+
+    // Flag table for deletion.
+    VERBOSE(*_workload, "Flagging pending delete for: " << table_itr->first);
+    std::cerr << "Flagging pending delete for: " << table_itr->first << std::endl;
+    icontext->_dyn_table_runtime[table_itr->second]._pending_delete = true;
+
+    // Add table to the pending delete list.
+    ASSERT(std::find(pending_delete.begin(), pending_delete.end(), table_itr->first) ==
+      pending_delete.end());
+    pending_delete.push_back(table_itr->first);
```

Comment:
This code is repeated below, please refactor with a function.

Reply:
Done! See [2be2f3b](https://github.com/wiredtiger/wiredtiger/pull/8798/commits/2be2f3b5ca7117dd1299929a57a10dae0a33bea1).